### PR TITLE
[FLINK-14198][python] Add "type" and "rtype" options to flink python API docstrings of table.py and table_environment.py

### PR DESCRIPTION
**Pull request by WeiZhong94**
## What is the purpose of the change

*This pull request adds "type" and "rtype" options to  all the API docstrings of table.py and table_environment.py.*

## Brief change log

  - *Add "type" and "rtype" options to  all the API docstrings of table.py and table_environment.py.*

## Verifying this change

This change is a trivial rework / code cleanup without any test coverage.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (no)
  - The serializers: (no)
  - The runtime per-record code paths (performance sensitive): (no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Yarn/Mesos, ZooKeeper: (no)
  - The S3 file system connector: (no)

## Documentation

  - Does this pull request introduce a new feature? (no)
  - If yes, how is the feature documented? (not applicable)

### CODE REVIEW

File: flink-python/pyflink/table/table_environment.py [hunk 1]
```diff
@@ -539,6 +571,7 @@ def use_catalog(self, catalog_name):
         +----------------+-----------------------------------------+
 
         :param: catalog_name: The name of the catalog to set as the current default catalog.
```
Comment (HuangXingBo) [inline flink-python/pyflink/table/table_environment.py]: Need to remove the ':' after param, or else the param str will not be parsed by sphix

File: flink-python/pyflink/table/table_environment.py [hunk 2]
```diff
@@ -602,6 +636,7 @@ def use_database(self, database_name):
         .. seealso:: :func:`~pyflink.table.TableEnvironment.use_catalog`
 
         :param: database_name: The name of the database to set as the current database.
```
Comment (HuangXingBo) [inline flink-python/pyflink/table/table_environment.py]: ditto

File: flink-python/pyflink/table/table_environment.py (hunk 1 above)
Comment (WeiZhong94) [reply flink-python/pyflink/table/table_environment.py]: Good catch!

**Comment by WeiZhong94** [comment]
@HuangXingBo Thanks for your review! I have removed the unnecessary colons in the latest commit.

**Comment by sunjincheng121** [comment]
@flinkbot approve description consensus architecture

File: flink-python/pyflink/table/table.py [hunk 3]
```diff
@@ -538,8 +579,10 @@ def over_window(self, *over_windows):
 
             Over-windows for batch tables are currently not supported.
 
-        :param over_windows: :class:`OverWindow`s created from :class:`Over`.
-        :return: A :class:`OverWindowedTable`.
+        :param over_windows: over windows created from :class:`Over`.
```
Comment (sunjincheng121) [inline flink-python/pyflink/table/table.py]: class: `Over` -> class:`pyflink.table.window.Over`  ?

File: flink-python/pyflink/table/table.py [hunk 4]
```diff
@@ -621,6 +672,7 @@ def insert_into(self, table_path):
 
         :param table_path: The path of the registered :class:`TableSink` to which
```
Comment (sunjincheng121) [inline flink-python/pyflink/table/table.py]: I prefer using full path in `:class:` tag, even `class:TableSink` is work well. What do you think?

**Comment by WeiZhong94** [comment]
@sunjincheng121 Thanks for your review! I have addressed your comments in the latest commit.

**Comment by sunjincheng121** [comment]
@flinkbot approve all

**Pull request closed without merging**